cmd/notation/login.go

```diff
@@ -70,61 +73,69 @@ func runLogin(ctx context.Context, opts *loginOpts) error {
 	// input username and password by prompt
 	reader := bufio.NewReader(os.Stdin)
 	var err error
-	if opts.Username == "" {
-		opts.Username, err = readUsernameFromPrompt(reader)
-		if err != nil {
-			return err
-		}
-	}
-
 	if opts.Password == "" {
-		opts.Password, err = readPasswordFromPrompt(reader)
+		var isToken bool
+		if opts.Username == "" {
+			opts.Username, err = readUsernameFromPrompt(reader)
+			if err != nil {
+				return err
+			}
+			if opts.Username == "" {
+				// the username is empty, the password is used as a token
+				isToken = true
+			}
+		}
+		opts.Password, err = readPasswordFromPrompt(reader, isToken)
 		if err != nil {
 			return err
 		}
+		if opts.Password == "" {
+			if isToken {
+				return errors.New("token required")
+			}
+			return errors.New("password required")
+		}
 	}
+	cred := opts.Credential()
 
-	if err := validateAuthConfig(ctx, opts, serverAddress); err != nil {
-		return err
+	credsStore, err := auth.NewCredentialsStore()
+	if err != nil {
+		return fmt.Errorf("failed to get credentials store: %v", err)
 	}
-
-	nativeStore, err := auth.GetCredentialsStore(ctx, serverAddress)
+	registry, err := getRegistryLoginClient(ctx, &opts.SecureFlagOpts, serverAddress)
 	if err != nil {
-		return fmt.Errorf("could not get the credentials store: %v", err)
+		return fmt.Errorf("failed to get registry client: %v", err)
 	}
+	if err := credentials.Login(ctx, credsStore, registry, cred); err != nil {
+		registryName := registry.Reference.Registry
+		if !errors.Is(err, credentials.ErrPlaintextPutDisabled) {
+			return fmt.Errorf("failed to log in to %s: %v", registryName, err)
+		}
 
-	// init creds
-	creds := newCredentialFromInput(
-		opts.Username,
-		opts.Password,
-	)
-	if err = nativeStore.Store(serverAddress, creds); err != nil {
-		return fmt.Errorf("failed to store credentials: %v", err)
+		// ErrPlaintextPutDisabled returned by Login() indicates that the
+		// credential is validated but is not saved because there is no native
+		// credentials store available
+		if savedCred, err := credsStore.Get(ctx, registryName); err != nil || savedCred != cred {
+			if err != nil {
+				// if we fail to get the saved credential, log a warning
+				// but do not throw the GET error, as the error could be
+				// confusing to users
+				logger := log.GetLogger(ctx)
+				logger.Warnf("Failed to get the existing credentials for %s: %v", registryName, err)
+			}
+			return fmt.Errorf("failed to log in to %s: the credential could not be saved because a credentials store is required to securely store the password. See %s",
```

Comment:
Question: when `err == nil && savedCred != cred`, the current error message is `...the credential could not be saved because a credentials store is required to securely store the password...` But I think the correct error message should be something like `...user provided credential does not match with the existing one...`?

Reply:
Comparing the existing credential with the provided one is a trick - If the credentials store is not set up, but there is an identical existing credential, the user can still do signing.
If the provided credential is different, we cannot save it because the credentials store is not available, and the login fails. 
A message like "user provided credential does not match with the existing one..." does not make sense to the user.

Reply:
Discussed offline with @Wwwsylvia, we may update this error message in the future.